frigate/record/maintainer.py

```diff
@@ -265,10 +265,16 @@ async def validate_and_move_segment(
                     self.end_time_cache.pop(cache_path, None)
         # else retain days includes this segment
         else:
-            record_mode = self.config.cameras[camera].record.retain.mode
-            return await self.move_segment(
-                camera, start_time, end_time, duration, cache_path, record_mode
-            )
+            most_recently_processed_frame_time = self.object_recordings_info[camera][
+                -1
+            ][0]
+
+            # ensure delayed segment info does not lead to lost segments
+            if most_recently_processed_frame_time >= start_time.timestamp():
```

Comment:
Shouldn't this be `end_time`?

Reply:
That would make sense to me, but were you thinking of something different, @NickM-27?